pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -388,85 +377,25 @@ func fixLibvirtCrcNetworkActive() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	logging.Debug("libvirt 'crc' network started")
-	return true, nil
-}
-
-func checkCrcDnsmasqConfigFile() (bool, error) {
-	logging.Debug("Checking dnsmasq configuration")
-	c := []byte(crcDnsmasqConfig)
-	_, err := os.Stat(crcDnsmasqConfigPath)
-	if err != nil {
-		return false, fmt.Errorf("File not found: %s: %s", crcDnsmasqConfigPath, err.Error())
-	}
-	config, err := ioutil.ReadFile(filepath.Clean(crcDnsmasqConfigPath))
-	if err != nil {
-		return false, fmt.Errorf("Error opening file: %s: %s", crcDnsmasqConfigPath, err.Error())
-	}
-	if !bytes.Equal(config, c) {
-		return false, fmt.Errorf("Config file contains changes: %s", crcDnsmasqConfigPath)
-	}
-	logging.Debug("dnsmasq configuration is good")
-	return true, nil
-}
-
-func fixCrcDnsmasqConfigFile() (bool, error) {
-	logging.Debug("Fixing dnsmasq configuration")
-	err := crcos.WriteToFileAsRoot(
-		fmt.Sprintf("write dnsmasq configuration in %s", crcDnsmasqConfigPath),
-		crcDnsmasqConfig,
-		crcDnsmasqConfigPath,
-	)
-	if err != nil {
-		return false, fmt.Errorf("Failed to write dnsmasq config file: %s: %v", crcDnsmasqConfigPath, err)
-	}
-
-	logging.Debug("Reloading NetworkManager")
-	sd := systemd.NewHostSystemdCommander()
-	if _, err := sd.Reload("NetworkManager"); err != nil {
-		return false, fmt.Errorf("Failed to restart NetworkManager: %v", err)
-	}
-
-	logging.Debug("dnsmasq configuration fixed")
-	return true, nil
-}
-
-func checkCrcNetworkManagerConfig() (bool, error) {
-	logging.Debug("Checking NetworkManager configuration")
-	c := []byte(crcNetworkManagerConfig)
-	_, err := os.Stat(crcNetworkManagerConfigPath)
-	if err != nil {
-		return false, fmt.Errorf("File not found: %s: %s", crcNetworkManagerConfigPath, err.Error())
-	}
-	config, err := ioutil.ReadFile(filepath.Clean(crcNetworkManagerConfigPath))
+	// Set DNS on the network
+	cmd = exec.Command("nmcli", "connection", "modify", "crc",
+		"ipv4.dns", "192.168.130.11",
+		"ipv4.dns-priority", "10")
+	buf = new(bytes.Buffer)
+	cmd.Stderr = buf
+	err = cmd.Run()
 	if err != nil {
-		return false, fmt.Errorf("Error opening file: %s: %s", crcNetworkManagerConfigPath, err.Error())
-	}
-	if !bytes.Equal(config, c) {
-		return false, fmt.Errorf("Config file contains changes: %s", crcNetworkManagerConfigPath)
+		return false, fmt.Errorf("%v : %s", err, buf.String())
 	}
-	logging.Debug("NetworkManager configuration is good")
-	return true, nil
-}
-
-func fixCrcNetworkManagerConfig() (bool, error) {
-	logging.Debug("Fixing NetworkManager configuration")
-	err := crcos.WriteToFileAsRoot(
-		fmt.Sprintf("write NetworkManager config in %s", crcNetworkManagerConfigPath),
-		crcNetworkManagerConfig,
-		crcNetworkManagerConfigPath,
-	)
+	// We need to do this to get NetworkManager to set the DNS in the resolv.conf
+	cmd = exec.Command("nmcli", "connection", "up", "crc")
```

Comment:
After running this with sudo, the VM becomes unpingable from the host, so this has more consequences on host networking than adding a 192.168.130.11 entry to /etc/resolv.conf

Reply:
using goodhosts sounds like a non-alternative for every plaform?

Reply:
how does it get around the permissions for adding entries?

Reply:
grant permissions as part of the setup?
see macOS solution

Reply:
i don't see the advantage then. I understood that the reason we want to drop dnsmasq is move towards rootless access but both the current dnsmasq setup and your proposed solution requires privileged access at setup time *only*. No?

Reply:
your proposed solution is less intrusive for sure though so that reason might be good enough to persue it.

Reply:
configuring network manager to use dnsmasq (as we do now) feels better to me than giving write permissions to the current user to random files in /etc

Reply:
@cfergeau I agree but I understood that it will be only a setup-time thing. i-e `crc setup` adding the needed hostnames and elevate permissions just of that. Although `crc delete` will need to undo this(?) so it'll also need to elevate permissions.

Reply:
Ah right, the IP is fixed on linux, so we don't have to make changes every time the VM starts contrary to what happens on macos.
However, new hostnames are dynamically created when new routes are created in openshift, so we still need some kind of dynamic dns, so I'm not sure how something based on /etc/hosts would work. On macos, we only need to change it for 'oc', which only need 2 or 3 static hostnames to be set.

Reply:
> configuring network manager to use dnsmasq (as we do now) feels better to me

I do not believe we should pursue getting rid of dnsmasq, but to get to 'rootless' where possible... or at least to a point this is the least needed (preferably only during setup).

note: about macOS, it does NOT remove the entries, and this to me feel bad too.